oap-server/server-receiver-plugin/skywalking-trace-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/trace/provider/parser/SegmentParseV2.java

```diff
@@ -82,10 +82,11 @@ public boolean parse(BufferData<UpstreamSegment> bufferData, SegmentSource sourc
 
             List<UniqueId> traceIds = upstreamSegment.getGlobalTraceIdsList();
 
+            SegmentObject segmentObject = parseBinarySegment(upstreamSegment);
```

Comment:
`SegmentParseV2#parse` will be invoked multiple times, due to `SegmentParseV2#Producer#call` triggered by `DataStreamReader#L159-L161`. So, the `#parseBinarySegment` should be called only when `bufferData.getV2Segment() == null` from my perspective

Reply:
After the debug, as you said, read from the file and go back to this judgment.